ratatui-widgets/examples/calendar.rs

```diff
@@ -0,0 +1,97 @@
+//! library you are using.
+//!
+//! [Ratatui]: https://github.com/ratatui/ratatui
+//! [widget examples]: https://github.com/ratatui/ratatui/blob/main/ratatui-widgets/examples
+//! [examples readme]: https://github.com/ratatui/ratatui/blob/main/examples/README.md
+
+use color_eyre::Result;
+use ratatui::{
+    crossterm::event::{self, Event},
+    layout::{Constraint, Layout, Rect},
+    style::{Color, Modifier, Style, Stylize},
+    text::{Line, Span},
+    widgets::{Block, Padding},
+    DefaultTerminal, Frame,
+};
+use ratatui_widgets::calendar::{CalendarEventStore, Monthly};
+use time::{Date, Month, OffsetDateTime};
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+    let terminal = ratatui::init();
+    let result = run(terminal);
+    ratatui::restore();
+    result
+}
+
+/// Run the application.
+fn run(mut terminal: DefaultTerminal) -> Result<()> {
+    loop {
+        terminal.draw(draw)?;
+        if matches!(event::read()?, Event::Key(_)) {
+            break Ok(());
+        }
+    }
+}
+
+/// Draw the UI with 2 monthly calendars side by side.
+fn draw(frame: &mut Frame) {
+    let vertical = Layout::vertical([Constraint::Length(1), Constraint::Fill(1)]).spacing(1);
+    let horizontal = Layout::horizontal([Constraint::Percentage(50); 2]).spacing(1);
+    let [top, main] = vertical.areas(frame.area());
+    let [left, right] = horizontal.areas(main);
+
+    let title = Line::from_iter([
+        Span::from("Calendar Widget").bold(),
+        Span::from(" (Press 'q' to quit)"),
+    ]);
+    frame.render_widget(title.centered(), top);
+
+    render_current_month(frame, left);
+    render_styled_month(frame, right);
+}
+
+/// Render the current month calendar.
+fn render_current_month(frame: &mut Frame, area: Rect) {
+    let date = OffsetDateTime::now_utc().date();
+
+    let monthly = Monthly::new(date, CalendarEventStore::today(Style::new().red().bold()))
+        .block(Block::new().padding(Padding::new(0, 0, 2, 0)))
+        .show_month_header(Style::new().bold())
+        .show_weekdays_header(Style::new().italic());
+    frame.render_widget(monthly, area);
+}
+
+/// Render an arbitrary month with more styles.
+fn render_styled_month(frame: &mut Frame, area: Rect) {
+    // Release date of the movie Ratatouille.
+    let date = Date::from_calendar_date(2007, Month::June, 29).unwrap();
+
+    let mut event_store = CalendarEventStore::today(Style::new().red().bold());
+    event_store.add(date, Style::new().blue().italic());
+
+    let monthly = Monthly::new(date, event_store)
+        .show_surrounding(Style::default().add_modifier(Modifier::DIM))
+        .show_month_header(Style::new().bold())
+        .show_weekdays_header(
+            Style::default()
+                .add_modifier(Modifier::BOLD)
+                .fg(Color::Green),
+        )
+        .default_style(
+            Style::default()
+                .add_modifier(Modifier::BOLD)
+                .bg(Color::Rgb(50, 50, 50)),
+        );
```

Comment:
nits:
- I tend to extract variables whenever things like this wrap to new lines (e.g. `let default_style = ...` as it ends up saving a few vertical lines and marginally improving the readability.
- style fields that take `Into<Style>` where you're just passing a color or modifier can be passed as that color / modifier directly.
- Use the stylize stuff generally for shorter code.

Reply:
c1786c706372984fb67e9ee786f3c34eccedf1b6